src/templates/tag.js

```diff
@@ -1,36 +1,53 @@
 const {html} = require("common-tags");
 const {wrapper, renderMarkdown, metabox, alert, anchor} = require("./shared");
 
-function expandStructs(struct, tags) {
-  const structs = struct.fields
+function expandStructs(parentedStruct, tags) {
+  const {struct, parentName} = parentedStruct;
+  const results = struct.fields
     .filter(field => field.type == "TagReflexive" &&
       field.struct != "PredictedResource"
     )
-    .map(field => tags[field.struct]);
+    .map(field => ({struct: tags[field.struct], parentName}));
   if (struct.inherits) {
-    structs.push(tags[struct.inherits]);
+    results.push({
+      struct: tags[struct.inherits],
+      parentName: struct.inherits
+    });
   }
-  return structs;
+  return results;
 }
```

Comment:
These are code changes needed to track the name of the parent struct the tag dependency comes from.